oncogene.js

```diff
@@ -213,8 +213,41 @@ class Oncogene {
             }
         })
 
-        if (!(options.steps instanceof Array))
+        if (!(this.isArray(options.steps)))
             throw new Error('Steps should be an array')
+
+        options.steps.forEach(step => {
+            if (true
+                && this.isUndefined(step.key)
+                && this.isUndefined(step.callback))
+                    throw new Error('Step should have key or callback')
+
+            if (true
+                && !this.isUndefined(step.callback)
+                && !this.isFunction(step.callback)) 
+                    throw new Error('Step callback should be a function')
+
+            if (true
+                && !this.isUndefined(step.key)
+                && !this.isString(step.key))
```

Comment:
It can be object with `toString`. It can be number.
`isUndefined` is enough.

(Actually it can be even undefined, but it's very strangely)